juriscraper/opinions/united_states/state/colo.py

```diff
@@ -84,43 +94,48 @@ def _extract_cases_from_sub_page(self, html_tree, date_obj):
             self.cases.append({
                 'date': date_obj,
                 'status': 'Published',
-                'name': self._extract_name_from_anchor(anchor),
+                'name': self._extract_name_from_text(text),
                 'url': self._extract_url_from_anchor(anchor),
                 'docket': self._extract_docket_from_text(text),
                 'citation': self._extract_citation_from_text(text),
                 'summary': self._extract_summary_relative_to_anchor(anchor),
                 'nature': self._extract_nature_relative_to_anchor(anchor),
             })
 
-    @staticmethod
-    def _extract_docket_from_text(text):
-        return text.split('No. ')[1].split('.')[0].rstrip('.').replace(' &', ',')
+    @classmethod
+    def _extract_docket_from_text(cls, text):
+        try:
+            match = re.match(cls.regex, text).group(6)
+        except:
+            raise InsanityException('Unable to parse docket from "%s"' % text)
+        dockets_raw = match.rstrip('.').replace('&', ' ').replace(',', ' ')
+        dockets = dockets_raw.split()
+        return ', '.join(dockets)
 
-    @staticmethod
-    def _extract_citation_from_text(text):
+    @classmethod
+    def _extract_name_from_text(cls, text):
+        try:
+            match = re.match(cls.regex, text).group(12)
+        except:
+            raise InsanityException('Unable to parse case name from "%s"' % text)
+        return match.strip().rstrip('.')
+
+    @classmethod
+    def _extract_citation_from_text(cls, text):
         return text.split('.')[0].strip()
 
-    @staticmethod
-    def _extract_text_from_anchor(anchor):
+    @classmethod
```

Comment:
If you're not using `cls`, you can annotate these as `@staticmethod`, which will allow you to drop the first argument.